commons/res/xml/preferences.xml

```diff
@@ -7,6 +7,11 @@
         android:title="@string/preference_license"
         android:defaultValue="CC BY-SA"
         />
+    <CheckBoxPreference
+        android:title="Automatically get current location"
+        android:defaultValue="false"
+        android:summary="Retrieve current location to offer category suggestions if image is not geotagged"
+        android:key="allowGps" />
```

Comment:
I recommend storing the allowGps string as a string resource and referencing it from XML as `@string/allow_gps` and code as `R.string.allow_gps`.